eth/downloader/downloader.go

```diff
@@ -304,6 +310,13 @@ func (d *Downloader) synchronise(id string, hash common.Hash, td *big.Int, mode
 		default:
 		}
 	}
+	for empty := false; !empty; {
```

Comment:
If a sync cycle fails, there might be leftovers in the headerProcCh since it's buffered. This cycle just cleans out any previous data left inside it before starting the new cycle.